flink-staging/flink-hcatalog/src/test/scala/org/apache/flink/hcatalog/scala/test/HCatInputOutputFormatITest.scala

```diff
@@ -0,0 +1,215 @@
+      "fields terminated by '\t' " +
+      "COLLECTION ITEMS TERMINATED BY '/' " +
+      "MAP KEYS TERMINATED BY ':' " +
+      "STORED AS TEXTFILE"
+    driver.run("drop table test_table")
+    val retCode1: Int = driver.run(createTable).getResponseCode
+    assertTrue(retCode1 == 0)
+    val partitionValues: Map[String, String] = Map("c0" -> "part0")
+    val opf: HCatOutputFormat[(List[String], Map[Int, String], List[Any], String, Float)] =
+      new HCatOutputFormat[(List[String], Map[Int, String], List[Any], String, Float)](null,
+        "test_table", partitionValues, hiveConf)
+
+    val t1 = (List[String]("a1","b1","c1"), Map[Int, String](1 -> "v11", 2 -> "v12"),
+      List[Any]("d", 1), "e1", 0.1f)
+    val t2 = (List[String]("a2","b2","c2"), Map[Int, String](1 -> "v21", 2 -> "v22"),
+      List[Any]("d", 2), "e2", 0.2f)
+    val l: List[(List[String], Map[Int, String], List[Any], String, Float)]=
+      List[(List[String], Map[Int, String], List[Any], String, Float)](t1, t2)
+    val env: ExecutionEnvironment = ExecutionEnvironment.createCollectionsEnvironment
+    val d: DataSet[(List[String], Map[Int, String], List[Any], String, Float)] =
+      env.fromCollection(l)
+    d.output(opf)
+    env.execute
+    val outputFileName: String = makePathASafeFileName(warehouseDir +
+      File.separator + "test_table/c0=part0/part-m-00001")
+    val reader: BufferedReader = new BufferedReader(new FileReader(outputFileName))
+    try {
+      var line: String = reader.readLine
+      assertEquals("1st row", line, "a1/b1/c1\t1:v11/2:v12\td/1\te1\t0.1")
+      line = reader.readLine
+      assertEquals("2nd row", line, "a2/b2/c2\t1:v21/2:v22\td/2\te2\t0.2")
+      reader.close
+    }
+    catch {
+      case e: Exception => {
+        e.printStackTrace
+      }
+    }
+  }
+
+  @Test
+  @throws(classOf[Exception])
+  def testTypeInfoCheck {
+    val createTable: String = "CREATE TABLE test_table(" +
+      "c1 array<string>,\n" +
+      "c2 map<int,string>,\n" +
+      "c3 struct<name:string,score:int>,\n" +
+      "c4 string,\n" +
+      "c5 float)\n" +
+      "partitioned by (c0 string)\n" +
+      "row format delimited " +
+      "fields terminated by '\t' " +
+      "COLLECTION ITEMS TERMINATED BY '/' " +
+      "MAP KEYS TERMINATED BY ':' " +
+      "STORED AS TEXTFILE"
+    driver.run("drop table test_table")
+    val retCode1: Int = driver.run(createTable).getResponseCode
+    assertTrue(retCode1 == 0)
+    val partitionValues: Map[String, String] = Map("c0" -> "part0")
+
+    //wrong type, should throw exception
+    try {
+      val opf: HCatOutputFormat[(List[String], Map[Int, String], List[Any], Float, Float)] =
+        new HCatOutputFormat[(List[String], Map[Int, String], List[Any], Float, Float)](null,
+          "test_table", partitionValues, hiveConf)
+    } catch{
+      case ioe:IOException => assertEquals(ioe.getMessage, "field has different type from required")
+      case _ => fail()
+    }
```

Comment:
This is not nice solution to check raising exception. Add an `expected` parameter into `@Test` annotation and remove try-catch block. The example is following: `@Test(expected = IOException.class)`.

Checking message of exception is bad. If you have to specify the thrown exception, choose other exception such as `InvalidTypesException`.